File: src/daily_report/renderers/markdown.py
from __future__ import annotations

from typing import Any


def render_markdown(report: dict[str, Any]) -> str:
    lines = [f"# {report['title']}", "", report.get("summary", ""), ""]
    for section in report.get("sections", []):
        lines += [f"## {section['name']}", "", section.get("summary", ""), ""]
        for item in section.get("items", []):
            lines.append(f"### [{item['repo']}]({item['url']})")
            lines.append("")
            lines.append(item.get("why_it_matters", ""))
            lines.append("")
            highlights = item.get("highlights", [])
            for h in highlights:
                lines.append(f"- {h}")
            lines.append("")
    return "\n".join(lines).strip() + "\n"

File: src/daily_report_js/renderers/markdown.js
import { formatNum } from '../utils/text.js';

export function renderMarkdown(report) {
  const section = report.section || 'github';
  if (section === 'media') {
    const lines = [
      `# ${report.title || 'Media Report'}`,
      '',
      report.summary || '',
      '',
    ];
    for (const item of report.items || []) {
      const pub = item.published_at ? String(item.published_at).slice(0, 10) : '';
      lines.push(`### [${item.title || ''}](${item.url || ''})`);
      lines.push(`*${item.source || ''}* · ${pub}`);
      lines.push('');
      lines.push(`${item.title || ''}：${item.summary || item.impact_and_outlook || item.why_it_matters || ''}`);
      lines.push('');
    }
    return `${lines.join('\n').trim()}\n`;
  }

  if (section === 'x') {
    const lines = [
      `# ${report.title || 'X Trending'}`,
      '',
      report.summary || '',
      '',
    ];
    for (const item of report.items || []) {
      const author = item.author_handle ? `@${item.author_handle}` : item.author_name || '';
      lines.push(`### [${author}](${item.url || ''})`);
      lines.push(`*👁 ${formatNum(item.view_count || 0)} · ❤️ ${formatNum(item.like_count || 0)}*`);
      lines.push('');
      lines.push(item.text || '');
      lines.push('');
    }
    return `${lines.join('\n').trim()}\n`;
  }

  const lines = [
    `# ${report.title || 'GitHub Trending'}`,
    '',
  ];
  for (const item of report.items || []) {
    const total = Number(item.stars_total || 0).toLocaleString();
    const today = Number(item.stars_today || 0).toLocaleString();
    const desc = item.description ? ` — ${item.description}` : '';
    lines.push(`\`${item.repo || ''}\` ⭐ ${total} (+${today})${desc}`);
    lines.push(`> ${item.url || ''}`);
    lines.push('');
  }
  return `${lines.join('\n').trim()}\n`;
}
